[Bug] `nix develop && make test` fails because `CGO_ENABLED=0` is incompatible with `go test -race`

### Is this a support request?

- [x] This is not a support request

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

When following the [Nix Development Workflow](https://github.com/juanfont/headscale#development-workflow):

```console
sfllaw@h2co3:~/Projects/headscale/main$ nix develop
sfllaw@h2co3:~/Projects/headscale/main$ echo $CGO_ENABLED
0
sfllaw@h2co3:~/Projects/headscale/main$ make test
Running Go tests...
go test -race ./...
go: -race requires cgo; enable cgo by setting CGO_ENABLED=1
make: *** [Makefile:48: test] Error 2
```

This seems to be caused because 75e24de7bd8d1b65ac40b6ea9090b0b979cf98dc added `CGO_ENABLED=0` to the `shellHook` for `nix develop`:
https://github.com/juanfont/headscale/blob/cfd845cb53efb2f953ff30c7d2ee70329b74f1ce/flake.nix#L265-L268

### Expected Behavior

```console
sfllaw@h2co3:~/Projects/headscale/main$ make test
Running Go tests...
go test -race ./...
```

I _think_ we should be running `CGO_ENABLED=1 go test -race ./...`:
https://github.com/juanfont/headscale/blob/cfd845cb53efb2f953ff30c7d2ee70329b74f1ce/Makefile#L44-L48

### Steps To Reproduce

See above.

### Environment

```markdown
- OS: Ubuntu 26.04
- Headscale version: cfd845cb53efb2f953ff30c7d2ee70329b74f1ce
- Tailscale version: n/a
```

### Runtime environment

- [ ] Headscale is behind a (reverse) proxy
- [ ] Headscale runs in a container

### Debug information

N/A

--- a/Makefile
+++ b/Makefile
@@ -45,7 +45,7 @@ build: check-deps $(GO_SOURCES) go.mod go.sum
 .PHONY: test
 test: check-deps $(GO_SOURCES) go.mod go.sum
 	@echo "Running Go tests..."
-	go test -race ./...
+	CGO_ENABLED=1 go test -race ./...
 
 
 # Formatting targets

